### ISSUE TITLE
Mailchimp Integration With Current Form Blocks Available

### ISSUE THREAD

**Issue by amanbhattarai4400** [reflexjs/reflex#35]
Recently, I tried to integrate MailChimp API to already existing form block from ReflexJS in my blog, however I'm not able to display success and error message on Submission.

Here's what I've done so far.

1) Installed the plugin and setup in `gatsby-config.js`

2) Created a New Subscribe Component in `src/@reflexjs/components`. Here's my `Subscribe.js` file:

```js
import addToMailchimp from "gatsby-plugin-mailchimp"
import React from "react"
import { Flexbox, P, Form, Label, Input, Button, VisuallyHidden, H4, Div } from "@reflexjs/components"


export default class Subscribe extends React.Component {
  constructor() {
    super()
    this.state = { name: '', email: '', result: null }
  }

  handleSubmit = async (e) => {
    const result = await addToMailchimp(this.state.email, { FNAME: this.state.name })
    if (result.result === 'error') {
      alert(`Whoops, ${this.state.name} you're already subscribed!`)
    } else {
      alert(`Thank you for subscribing ${this.state.name}!`)
    }
    this.setState({ result: result })
  }

  handleEmailChange = (event) => {
    this.setState({ email: event.target.value })
  }

  handleNameChange = (event) => {
    this.setState({ name: event.target.value })
  }

render() {
    return (
    <Flexbox flexDirection="column" textAlign="center">
      <Div p="7" mb="7" mt="7" borderRadius="2xl" boxShadow="3xl" bg="postblock"> 
      <H4>Sign up for The Best Bits</H4>
      <P fontSize="xl" mt="2" textAlign="center">
      A weekly round up of the best content from In Plain English, handpicked by our Founder
      </P>
      <Form d="grid" col="2|auto auto" gap="4" onFinish={this.handleSubmit}>
      <VisuallyHidden>
          <Label htmlFor="form-04-email">Email</Label>
          <Label htmlFor="form-04-name">Name</Label>
        </VisuallyHidden>
        <Input
          type="email"
          id="form-04-email"
          name="email"
          placeholder="Type your email"
          variant="md"
          onChange={this.handleEmailChange}
          rules={[{ required: true, message: 'Please input your email.' }]}
        />
        <Input
          type="name"
          id="form-04-name"
          name="name"
          placeholder="Type your name"
          variant="md"
          onChange={this.handleNameChange}
          rules={[{ required: true, message: 'Please input your name.' }]}
        />
        <Button
          variant="primary md"
          label="Submit"
          type="submit"
        > Join
        </Button>
      </Form>
      </Div>
      </Flexbox>
    )
  }
}
``` 

And I imported `import Subscribe from "../components/Subscribe"` and called `<Subscribe />` in my `post.js`.

Now, when I press Join Button, the values get appended to address, no result. I attempted with `gatsby build` as well, same result. Can you help me figure it out please?

**Comment by arshad** [comment]
is `addToMailchimp` not called at all?

Can you try with:

```diff
handleSubmit = async (e) => {
+ e.preventDefault()
```

**Comment by amanbhattarai4400** [comment]
Its working now, I removed the plugin and reinstalled everything with above change, and is now successfully delivering the details.

So, my other question is: 

Can I import and add it to a separate page (for instance, `/newsletter`)?

I'm getting path error while trying to import and use `<Subscribe />`. 

`Subscribe.js` is located in `src/@reflexjs/components` and `/newsletter` is a normal page.

**Comment by arshad** [comment]
You don't really need to place it inside `src/@reflexjs/components`. Your custom components can go inside `src`.

Place your file at `src/Subscribe.js`.

Then include it in your page as follows:

```jsx
---
title: Title of page
---

import Subscribe from "../../src/Subscribe"

<Subscribe />
```

**Comment by amanbhattarai4400** [comment]
Thanks, it worked :+1:
